waspc/packages/deploy/src/providers/fly/helpers/flyctlHelpers.ts

```diff
@@ -0,0 +1,72 @@
+import { exit } from 'process'
+import { $, question } from 'zx'
+import { silence, isYes, waspSays } from './helpers.js'
+
+export async function flyctlExists(): Promise<boolean> {
+  try {
+    await $`flyctl version`
+    return true
+  } catch {
+    return false
+  }
+}
+
+export async function isUserLoggedIn(): Promise<boolean> {
+  try {
+    await $`flyctl auth whoami`
+    return true
+  } catch {
+    return false
+  }
+}
+
+export async function ensureUserLoggedIn() {
```

Comment:
Well that is the nomenclature that I made up :D, so it is not like it is an accepted standard or something, but yeah would I usually do is:
1. prefix with `check` if it looks at something and returns Bool. Normally doesn't do a mutation.
2. prefix with `ensure` if it ensures that certain state is as it should be. It normally does a mutation only if needed, and doesn't return anything (it can throw error if it fails).

So for example if we have `checkIfUserLoggedIn`, then that would check that out and return true of false.
But if we have `ensureUserIsLoggedIn`, that would check if user is logged in, if yes it would just return;, if not it would log in the user and then return;. So whatever is the state of the external world, you can be sure that once this function is done, user is logged in.

But as I said I made this up, and it meant a lot dring the time when I was using it in a codebase that was in JS and we had no types. I still do like it though.

Reply:
Haha true. I do like the naming convention, I just forget sometimes. Ok, then we are on the same page here already. In these functions for `ensure*` I was making a check, and would exit on failure or just return on pass. 👍🏻